src/main/main.ino

```diff
@@ -263,6 +278,33 @@ void DoCount(){
   }
 }
 
+void PIDTesting(){
+  char outputBuffer[BufferLen]; //This is for printing values out to terminal
+  float target_lon=0; //Example value to be changed later
+  float target_lat=0; //Example value to be changed later
+
+  // Take these GPS and add x,y,z, from kalman filter
+  // Lat_Fast = GPS.Fast+X_Moved*ConversionFactor
+  int Radius = 6378100;
+  float lon_fast = sensorData_inst.gpsData.Longitude + X_Xaxis(0,0)*180/Radius/pi;
+  float lat_fast = sensorData_inst.gpsData.Latitude + X_Yaxis(0,0)*180/Radius/pi;
+  //Height = current GPS_Position & use kinematics to get new height?
+  
+  //Get bearing between two sets of coordinates (Euler Angle or using GPS bearings?)
+  double pv = sensorData_inst.imuData.EulerAngles.v2  //process variable from current position
+  //double pv = courseTo(lat_fast,lon_fast,target_lat,target_lon);
+
+  //compute PID angle using PID structure
+  int yaw = PID.PIDcalculate(pv);
+
+  //Send to servos
+  steering(yaw);
+  
+  //Print to terminal
+  sprintf(outputBuffer, "Pv: %.5lf \t Error: %.5lf \t Output: %.5lf\t Integral: %.5lf \t \n", pv, PID.error, yaw, PID.integral); 
+  SERIAL_PORT.print(outputBuffer);
+}
```

Comment:
I don't know if this works, we can test this on Saturday. **Question for you, did the code compile before you pushed it?**